src/alloc.rs

```diff
@@ -59,7 +59,7 @@ unsafe fn align_block(ptr: *mut u8, align: usize) -> *mut u8 {
         // we know that `offset >= POOL_ALIGNMENT` here. We then verify that `POOL_ALIGNMENT`
         // serves the needs of our `Marker` object. Note that all but the first assertion are
         // constant expressions, so the compiler will optimize them away.
-        let offset = align - (ptr as usize & (align - 1));
+        let offset = ptr.align_offset(align);
```

Comment:
I don't think this is correct. The `align_offset()` function only works on allocated blocks and allows implementations to return `usize::MAX`. Note that `library/std/src/sys/windows/alloc.rs` does something similar and it does **not** use `align_offset()` either.